docs/puzzles/day1.md

```diff
@@ -0,0 +1,139 @@
+---
+sidebar_position: 1
+---
+
+# Day 1: Sonar Sweep 
+by @adpi2
+
+## Puzzle description
+
+https://adventofcode.com/2021/day/1
+
+## Solution of Part 1
+
+The first step is to transform the input to a sequence of integers.
+We can do so with:
+```scala
+val depths: Seq[Int] = input.linesIterator.map(_.toInt).toSeq
+```
+
+The second step and hardest challenge of this puzzle is to compute all pairs of consecutive depth measurments.
+
+For each index `i` from 0 until `depths.size - 1` we can create a pair of the depths at index `i` and `i + 1`.
+```scala
+val pairs: Seq[(Int, Int)] =
+  for i <- 0 until depths.size -1
+  yield (depths(i), depths(i + 1))
+```
+
+Then we can count the pairs whose first element is smaller than its second element.
```

Comment:
Perhaps, before this sentence, show an example of input and what `pairs` is for that input. That should help people visualize what is happening.